implementations/rust/ockam/ockam_vault/src/service/models/common.rs

```diff
@@ -0,0 +1,47 @@
+use minicbor::{Decode, Encode};
+use ockam_core::compat::borrow::Cow;
+use ockam_core::vault::{PublicKey, SecretType};
+use ockam_core::{Error, Result};
+
+#[cfg(feature = "tag")]
+use ockam_core::TypeTag;
+
+#[derive(Debug, Clone, Encode, Decode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct PublicKeyModel<'a> {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<8922437>,
+    #[cbor(b(1), with = "minicbor::bytes")]
+    data: Cow<'a, [u8]>,
+    #[b(2)] stype: SecretType
+}
+
+impl<'a> PublicKeyModel<'a> {
+    pub fn new(data: impl Into<Cow<'a, [u8]>>, stype: SecretType) -> Self {
+        Self {
+            #[cfg(feature = "tag")]
+            tag: TypeTag,
+            data: data.into(),
+            stype,
+        }
+    }
+    pub fn data(&self) -> &Cow<'a, [u8]> {
+        &self.data
+    }
+    pub fn stype(&self) -> SecretType {
+        self.stype
+    }
+
+    pub fn from(public_key: &'a PublicKey) -> Self {
+        Self::new(Cow::from(public_key.data()), public_key.stype())
+    }
+}
+
+impl<'a> TryInto<PublicKey> for PublicKeyModel<'a> {
```

Comment:
I don't want to have additional functions on `PublicKey` related to this. In other words `PublicKeyModel` is dependent on `PublicKey`, but I want to avoid the opposite dependency